libminifi/include/utils/TimeUtil.h

```diff
@@ -70,6 +72,16 @@ class Clock {
  public:
   virtual ~Clock() = default;
   virtual std::chrono::milliseconds timeSinceEpoch() const = 0;
+  virtual std::chrono::milliseconds wait_until(std::condition_variable& cv, std::unique_lock<std::mutex>& lck, std::chrono::milliseconds time, const std::function<bool()>& pred) {
```

Comment:
based on the comment above `Clock`, these classes' purpose is to be mocked, as it is not possible to safely wait when using a `Clock` (as it might be an instance of `ManualClock`) I think this new method has relevance to all users of `Clock`